docs/token-api.md

```diff
@@ -109,23 +125,91 @@ It consists of three parts:
 
 - `Metadata`: It is a collection of `Token Metadata`, one entry for each Token Action.
   Parties, assembling a token request, exchange metadata that contain secret information used by
   the parties to check the content of the token actions. This is particularly relevant when using ZK-based drivers.
   Notice that, the ledger does not store any metadata.
 
-Looking ahead: As we mentioned earlier, a Token Request is itself agnostic to the details of the specific Blockchain.
-Indeed, a Token Request must be translated to the Transaction format of the target Blockchain to become meaningful.
+Looking ahead: As we mentioned earlier, a Token Request is itself agnostic to the details of the specific backend.
+Indeed, a Token Request must be translated to the Transaction format of the target backend to become meaningful.
 A service called `Token Request Translator` translates the token requests.
 The `Token Request Translator` does not belong to the Token API. It is offered as a service on top of the `Token API`
-because it is blockchain dependant. 
+because it is backend dependant. 
 
-Here is a pictorial representation of the translation process. 
+Here is a pictorial representation of the translation process for Fabric
 
 ![token_request_translator.png](imgs/token_request_translator.png)
 
 In Fabric, it is the `Token Chaincode` that performs validation and translations of token requests.
-In the section dedicated to [`What you need to build Token-Based Applications on top of Fabric`](./services.md), 
-we will learn more about this process.
+More information [`here`](./services.md).
+
+## Validator 
+
+The validator (`token.Validator`) is the component that sets the validation rules for a Token Request. The rules depend on the
+type of tokens supported (fungible, non-fungible, and so on), and on the specific driver implementation.
+A Validator validates Token Requests with the respect to:
+- A given Anchor (e.g., Fabric TxID), and
+- The Ledger. Notice that, in certain implementations, the ledger might not be needed.
+
+Even though, certain validation rules are driver specific, we can identify the following general validation rules.
+A token request should:
+- Well-formed, and
+- Satisfies the constraints of the payment system. Namely:
+- Only the `rightful owner` can transfer a token,
+- No token can be created out of the blue,
+- Audited(able)
+- Etc. (Each implementation can enforce additional requirements, if needed)
+
+## Vault
+
+The vault (`token.Vault`) gives access to the tokens that are owned by the wallets in the wallet manager.
+
+The developer can access the vault of a given TMS as follows:
+```go
+    vault := tms.Vault()
+```
+
+## Selector Manager
+
+A token selector (`token.Selector`) allows the developer to select tokens from the vault under certain conditions.
+For example, a token selector can be used to select tokens of a given type and for a given amount that are owned by a specific wallet,
```

Comment:
that are associated with a specific wallet

Reply:
doesn't `owned` better represent the relation?